File: frontend/src/types/checklist.ts
import type { DamagePoint } from "./damage";

export type ChecklistStatus = "entregue" | "devolvido";
export type ChecklistItemStatus = "ok" | "nao_ok";
export type NivelCombustivel = "1/4" | "2/4" | "3/4" | "4/4";

export interface ChecklistItemData {
  nome: string;
  status: ChecklistItemStatus | null;
}

export interface ChecklistResponse {
  id: number;
  placa: string;
  unidade: string;
  subunidade: string | null;
  motorista: string;
  matricula_motorista: string;
  quilometragem_inicial: number;
  status: ChecklistStatus;
  is_locked: boolean;
  itens: ChecklistItemData[] | null;
  nivel_combustivel: NivelCombustivel | null;
  data_entrega: string | null;
  avarias: DamagePoint[] | null;
  assinatura_responsavel: string | null;
  assinatura_motorista: string | null;
  quilometragem_final: number | null;
  data_devolucao: string | null;
  itens_devolucao: ChecklistItemData[] | null;
  nivel_combustivel_devolucao: NivelCombustivel | null;
  assinatura_responsavel_devolucao: string | null;
  assinatura_motorista_devolucao: string | null;
  observacoes: string | null;
  observacoes_devolucao: string | null;
  created_at: string;
}

File: backend/app/models/checklist.py
from datetime import datetime, timezone
from enum import Enum
from typing import Any

from sqlalchemy import JSON
from sqlmodel import Column, Field, SQLModel


class ChecklistStatus(str, Enum):
    entregue = "entregue"
    devolvido = "devolvido"


class Checklist(SQLModel, table=True):
    __tablename__ = "checklists"

    id: int | None = Field(default=None, primary_key=True)
    placa: str = Field(index=True)
    unidade: str
    subunidade: str | None = None
    motorista: str
    matricula_motorista: str
    quilometragem_inicial: float
    status: ChecklistStatus = Field(default=ChecklistStatus.entregue)
    is_locked: bool = Field(default=False)
    itens: list[Any] | None = Field(default=None, sa_column=Column(JSON))
    avarias: list[Any] | None = Field(default=None, sa_column=Column(JSON))
    nivel_combustivel: str | None = None
    data_entrega: datetime | None = None
    assinatura_responsavel: str | None = None
    assinatura_motorista: str | None = None
    quilometragem_final: float | None = None
    data_devolucao: datetime | None = None
    itens_devolucao: list[Any] | None = Field(default=None, sa_column=Column(JSON))
    nivel_combustivel_devolucao: str | None = None
    assinatura_responsavel_devolucao: str | None = None
    assinatura_motorista_devolucao: str | None = None
    observacoes: str | None = None
    observacoes_devolucao: str | None = None
    created_at: datetime = Field(
        default_factory=lambda: datetime.now(timezone.utc)
    )
